compiler/ml/types.ml

```diff
@@ -310,16 +310,14 @@ let same_record_representation x y =
   match x with
   | Record_regular -> y = Record_regular
   | Record_float_unused -> y = Record_float_unused
-  | Record_optional_labels lbls -> (
+  | Record_optional_labels -> (
     match y with
-    | Record_optional_labels lbls2 -> lbls = lbls2
+    | Record_optional_labels -> true
```

Comment:
The check if the labels are the same can be safely removed here?

Reply:
Yes there are no more labels represented here. Each label has its own optionality information.
Later on, we can probably remove the classification`Record_optional_labels` entirely.

Reply:
@cknitt take a look: removed `Record_optional_labels` entirely, now that the information is available directly from the fields.